esphome/components/deep_sleep/deep_sleep_component.cpp

```diff
@@ -13,12 +13,34 @@ static const char *const TAG = "deep_sleep";
 
 bool global_has_deep_sleep = false;  // NOLINT(cppcoreguidelines-avoid-non-const-global-variables)
 
+optional<uint32_t> DeepSleepComponent::get_run_duration_() const {
+#ifdef USE_ESP32
+  if (this->wakeup_cause_to_run_duration_.has_value()) {
+    esp_sleep_wakeup_cause_t wakeup_cause = esp_sleep_get_wakeup_cause();
+    switch (wakeup_cause) {
+      case ESP_SLEEP_WAKEUP_EXT0:
+      case ESP_SLEEP_WAKEUP_EXT1:
+        return this->wakeup_cause_to_run_duration_->gpio_cause;
+      case ESP_SLEEP_WAKEUP_TOUCHPAD:
+        return this->wakeup_cause_to_run_duration_->touch_cause;
+      default:
+        return this->wakeup_cause_to_run_duration_->default_cause;
+    }
+  }
+#endif
+  return this->run_duration_;
+}
+
 void DeepSleepComponent::setup() {
-  ESP_LOGCONFIG(TAG, "Setting up Deep Sleep...");
   global_has_deep_sleep = true;
 
-  if (this->run_duration_.has_value())
-    this->set_timeout(*this->run_duration_, [this]() { this->begin_sleep(); });
+  const optional<uint32_t> run_duration = get_run_duration_();
+  if (run_duration.has_value()) {
+    ESP_LOGCONFIG(TAG, "Scheduling Deep Sleep to start in %u ms", *run_duration);
+    this->set_timeout(*run_duration, [this]() { this->begin_sleep(); });
+  } else {
+    ESP_LOGCONFIG(TAG, "Deep Sleep not scheduled, run_duration is missing.");
+  }
```

Comment:
So to follow the convention, I'd suggest this: (+ restore the log message above, but GitHub doesn't allow suggestions on deletions :disappointed:)
```suggestion
  if (run_duration.has_value()) {
    ESP_LOGI(TAG, "Scheduling Deep Sleep to start in %u ms", *run_duration);
    this->set_timeout(*run_duration, [this]() { this->begin_sleep(); });
  } else {
    ESP_LOGD(TAG, "Not scheduling Deep Sleep, as no run duration is configured");
  }
```
(I also lowered the priority in the `else`-branch, as it's perfectly fine to not schedule a run duration).